core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt

```diff
@@ -217,24 +222,51 @@ fun buildFinalEnvelope(
         val sectionTime =
             getEnvelopeTimeAt(point.pathOffset) - getEnvelopeTimeAt(prevFixedPointOffset)
         val arrivalTime = prevFixedPointDepartureTime + sectionTime
-        var extraTime = point.arrival.seconds - arrivalTime
-        if (extraTime < 0.0) {
-            // TODO: raise a warning
-            standaloneSimLogger.warn("impossible scheduled point")
-            extraTime = 0.0
-        }
-        marginRanges.addAll(
-            distributeAllowance(
-                maxEffortEnvelope,
-                provisionalEnvelope,
-                extraTime,
-                margins,
-                prevFixedPointOffset,
-                point.pathOffset
+        val extraTime = point.arrival.seconds - arrivalTime
+        if (extraTime >= 0.0) {
+            marginRanges.addAll(
+                distributeAllowance(
+                    maxEffortEnvelope,
+                    provisionalEnvelope,
+                    extraTime,
+                    margins,
+                    prevFixedPointOffset,
+                    point.pathOffset
+                )
             )
-        )
+            prevFixedPointDepartureTime = arrivalTime + extraTime + (point.stopFor?.seconds ?: 0.0)
+        } else {
+            // We need to *remove* time compared to the provisional envelope.
+            // Ideally we would distribute the (negative) extra time following the same logic as
+            // when it's positive. But this is tricky: as we get closer to max effort envelope (hard
+            // limit), we need to redistribute the time in some cases.
+            // We currently handle this by ignoring the distribution over different margin ranges,
+            // we just set the time for the scheduled point without more details. It will be easier
+            // to handle it properly when we'll have migrated to standalone sim v3.
+            val maxEffortSectionTime =
+                getMaxEffortEnvelopeTimeAt(point.pathOffset) -
+                    getMaxEffortEnvelopeTimeAt(prevFixedPointOffset)
+            val earliestPossibleArrival = prevFixedPointDepartureTime + maxEffortSectionTime
+            var maxEffortExtraTime = point.arrival.seconds - earliestPossibleArrival
+            if (maxEffortExtraTime < 0.0) {
+                standaloneSimLogger.warn("impossible scheduled point")
+                // TODO: raise warning: scheduled point isn't possible
```

Comment:
I'm not sure what this TODO is, considering the above line

Reply:
The above line is just logging, we want that info to be forwarded to the user. It's related to https://github.com/OpenRailAssociation/osrd/issues/6925